src/test/scala/org/apache/spark/sql/pulsar/CachedPulsarClientBuilderSuite.scala

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.pulsar
+
+import org.apache.pulsar.client.api.ClientBuilder
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.pulsar.PulsarOptions.{AUTH_PARAMS, AUTH_PLUGIN_CLASS_NAME, TLS_ALLOW_INSECURE_CONNECTION, TLS_HOSTNAME_VERIFICATION_ENABLE, TLS_TRUST_CERTS_FILE_PATH}
+
+import java.{util => ju}
+
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{verify, when}
+
+import org.scalatest.mockito.MockitoSugar.mock
+
+class CachedPulsarClientBuilderSuite extends SparkFunSuite {
+  test("Should pass on authentication related parameters when authentication plugin is provided") {
```

Comment:
Could you add an integration test by using `TestContainer`.

Reply:
Currently the test suite is working locally due to multiple problems:
- "Too many open files" is getting thrown from some tests
- "Failed to remove cursor" warnings are thrown from PulsarMetadataReader

Please give me some time to fix these issues. If anything obvious comes up that might be an improvement to the documentation (so that others do not run into these problems) I'll add them to the documentation as well.

Reply:
Currently the test suite is working locally due to multiple problems:

- "Too many open files" is getting thrown from some tests (setting `docker-ulimits` to some very high value does not help) 
- "Failed to remove cursor" warnings are thrown from PulsarMetadataReader (albeit this might be normal, since the test otherwise succeeds).

Please give me some time to fix these issues. If you already know these and can share some insights on them, that would be most appreciated. I am not entirely sure whether these are caused by my modifications or is there something else misconfigured in my environment, but there are problems coming even when I just clone a clean master and do a `mvn clean install`. If anything comes up that might be an improvement to the documentation (so that others do not run into these problems) I'll add them to there as well.

There had been an other problem as well, a clean clone of the master gives back the following:
```
com.github.dockerjava.api.exception.ConflictException: {"message":"Conflict. The container name \"/standalone-standalone\" is already in use by container \"...long container hash...\". You have to remove (or rename) that container to be able to reuse that name."}
```
We can fix this by appending a sufficiently long random suffix to the name of the container. That made me pass this error but hit the others above.